src/common/selftest/selftest_firstlayer.cpp

```diff
@@ -0,0 +1,342 @@
+/**
+ * @file selftest_firstlayer.cpp
```

Comment:
note: Redundant to specify file name here if the documentation is in the same file.
I understand you like it that way so I don't insist on fixing that.